associations/models/election.py

```diff
@@ -0,0 +1,66 @@
+from collections import Counter
+from datetime import datetime, timezone
+
+from django.db import models
+from django.utils.functional import cached_property
+
+from associations.models.association import Association
+from authentication.models import User
+
+
+class Election(models.Model):
+    """
+        The Election class allows to carry a vote. It should always guarantee the anonymity of the voter.
+    """
+
+    id = models.AutoField(primary_key=True)
+
+    association = models.ForeignKey(Association, on_delete=models.CASCADE, related_name='elections')
+
+    name = models.CharField(max_length=200)
+
+    registered_voters = models.ManyToManyField(User,
+                                               related_name='allowed_elections',
+                                               help_text='User who are allowed to vote to this election.')
+    voters = models.ManyToManyField(User,
+                                    related_name='voted_elections',
+                                    help_text='People who have already voted to this election.')
+
+    starts_at = models.DateTimeField(null=False)
+    ends_at = models.DateTimeField(null=False)
+
+    max_choices_per_vote = models.PositiveIntegerField(default=1,
+                                                       help_text='The number of choices allowed per vote.')
+
+    @cached_property
+    def results(self):
+        # First, we fetch all the votes, but by replacing them by the associated choice name.
+        # We then wrap all of these choices into a Counter, which is a nice built-in object which will…
+        # count the votes for us.
+        return Counter([vote[0] for vote in self.votes.values_list('choices__name')])
+
+    @cached_property
+    def is_active(self):
+        return self.starts_at < datetime.now(tz=timezone.utc) < self.ends_at
+
+
+class Choice(models.Model):
+    """
+        Represents a choice in an election.
+    """
+
+    id = models.AutoField(primary_key=True)
+
+    election = models.ForeignKey(Election, on_delete=models.CASCADE, related_name='choices')
+    name = models.CharField(max_length=200)
+
+
+class Vote(models.Model):
```

Comment:
Vote is a bit ambiguous, as it can be the time when you vote or the bulletin

Reply:
Maybe `Election` -> `ElectionManager` or `AssociationElection` or `VogteBureau`and `Vote` -> `Election` ?

Reply:
Well, _election_ does not mean anything else than _élection_ in French (https://www.wordreference.com/enfr/election). But you are right, the English _vote_ does not have the same meaning as in French.

What about keeping `Election` for the whole process and `Ballot` for `Vote`?